pydra/engine/specs.py

```diff
@@ -90,6 +97,9 @@ def hash(self):
         """Compute a basic hash for any given set of fields."""
         from .helpers import hash_value, hash_function
 
+        # if inp_hash already calculated and nothing has changed, the old value can be used
+        if self.changed is False and self.inp_hash:
+            return self.inp_hash
         inp_dict = {}
         for field in attr_fields(self):
             if field.name in ["_graph_checksums", "bindings"] or field.metadata.get(
```

Comment:
shouldn't this if skip hashing `inp_hash` and `changed`. also attrs has a function called `on_setter` that you can use as well.

Reply:
it should

Reply:
i couldn't find `on_setter`, could you please point me to it